### PR TITLE
Add best practice

### PR DESCRIPTION
**Pull request by noahtalerman**
The `best-pratice/` folder now includes an example for the best practice when using Fleet and GitOps. This was discussed during MDM design review on 2023-08-29.

Still TODO after this PR is merged:

- Update the GitHub action to support this story: As a Fleet user, I want to clone the file structure, make a change to any file, and have this reflected in my Fleet so that I can use GitOps to manage Fleet. 
  - Noah: I added this to feature fest (2023-08-31)
- Use this repo for dogfood's queries, policies, controls, and settings.

### CODE REVIEW

**Comment by noahtalerman** [comment]
@lukeheath during design review (MDM) we decided on a best practice for using Fleet with GitOps.

This PR adds a template folder structure for the best practice. 

Soon, we'll update the GitHub action to support this template. Later, we'll update the template to instead use dogfood's queries, policies, controls, and settings. We can use this repo to dogfood the GitHub action.

cc @zhumo @rachaelshaw @mikermcneil

File: best-practice/README.md
Comment (lukeheath) [inline best-practice/README.md:18]: Nit: missing period at the end of the sentence for consistency.

File: best-practice/README.md
Comment (lukeheath) [inline best-practice/README.md:21]:
```suggestion
    - `<team name>.controls.yml` - controls config for a specific team, including configuration profiles, OS updates, setup experience, and scripts.
```

File: best-practice/README.md
Comment (lukeheath) [inline best-practice/README.md:18]:
```suggestion
- `teams` - folder for teams in Fleet. These folders contain team-level policies, queries, controls, and settings.
  - `<team name>` - folder for each team, with team-specific configs.
```

**Review by lukeheath — CHANGES_REQUESTED** [review]
Nice! Just a couple missing periods.

**Comment by noahtalerman** [comment]
Thanks Luke!

**Review by lukeheath — APPROVED** [review]

**Comment by noahtalerman** [comment]
@lukeheath can you please grant me write access to this repo? I can't merge the approved PR.

Maybe everyone at Fleet should have write access?

**Comment by noahtalerman** [comment]
@lukeheath yep! Merged. Thanks

**Pull request merged**
